OpenRA.Mods.Common/ServerTraits/LobbyCommands.cs

```diff
@@ -135,6 +135,10 @@ public bool InterpretCommand(S server, Connection conn, Session.Client client, s
 
 						client.Slot = s;
 						S.SyncClientToPlayerReference(client, server.MapPlayers.Players[s]);
+
+						var validatedColor = ColorValidator.ValidatePlayerColorAndGetAlternative(server, client.Color, client.Index, conn);
+						client.PreferredColor = client.Color = validatedColor;
```

Comment:
Sounds like somebody has broken how this feature is supposed to work, but I guess I can't penalize you for that.